clang/tools/driver/cc1_main.cpp

```diff
@@ -610,17 +611,34 @@ Expected<bool> ObjectStoreCachingOutputs::tryReplayCachedResult(
     const llvm::cas::CASID &ResultCacheKey) {
   DiagnosticsEngine &Diags = Clang.getDiagnostics();
 
-  Expected<Optional<llvm::cas::CASID>> Result = Cache->get(ResultCacheKey);
-  if (!Result)
-    return Result.takeError();
+  Optional<llvm::cas::CASID> Result;
+  {
+    llvm::ScopedDurationTimer ScopedTime([&Diags](double Seconds) {
+      SmallString<16> DurationStr;
+      llvm::raw_svector_ostream(DurationStr)
+          << llvm::format("%.6f", Seconds) << 's';
```

Comment:
Could you put the "s" into the format string and pass this directly to the diagnostic without an intermediate `raw_svector_ostream`?

Reply:
AFAICT I can't avoid the `raw_svector_ostream`, `StreamingDiagnostic` doesn't accept the format object.

Reply:
I've added an overload for `StreamingDiagnostic` to accept a format object.